app/uk/gov/hmrc/cdsreimbursementclaimfrontend/models/answers/package.scala

```diff
@@ -18,14 +18,19 @@ package uk.gov.hmrc.cdsreimbursementclaimfrontend.models
 
 import cats.data.NonEmptyList
 import uk.gov.hmrc.cdsreimbursementclaimfrontend.models.form.Duty
+import uk.gov.hmrc.cdsreimbursementclaimfrontend.models.ids.MRN
 import uk.gov.hmrc.cdsreimbursementclaimfrontend.models.reimbursement.DutyType
 import uk.gov.hmrc.cdsreimbursementclaimfrontend.models.upscan.UploadDocument
 
 package object answers {
 
-  type SupportingEvidencesAnswer = NonEmptyList[UploadDocument]
-  type DutiesSelectedAnswer      = NonEmptyList[Duty]
-  type ClaimsAnswer              = NonEmptyList[Claim]
+  type LeadMrn = MRN
+
+  type SupportingEvidencesAnswer               = NonEmptyList[UploadDocument]
+  type DutiesSelectedAnswer                    = NonEmptyList[Duty]
+  type ClaimsAnswer                            = NonEmptyList[Claim]
+  type AssociatedMRNsAnswer                    = NonEmptyList[MRN]
+  type AssociatedEntryDeclarationDetailsAnswer = NonEmptyList[EntryDeclarationDetails]
```

Comment:
I think you might be getting confused. EntryDeclarationDetails is for capturing the declaration details when in the Entry Number flow. For the journey you are doing, the information will be in DisplayDeclaration. You need a NonEmptyList[DisplayDeclaration]. The user is not going to provide you with that.